test/e2e/external/dataplane/connectivity.go

```diff
@@ -0,0 +1,96 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dataplane
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+)
+
+const (
+	extAppName = "ext-app"
+	extNetName = "pseudo-ext"
+
+	testImage         = "registry.access.redhat.com/ubi7/ubi:latest"
+	testContainerName = "ext-test-container"
+)
+
+var (
+	simpleHTTPServerCommand = []string{"python", "-m", "SimpleHTTPServer", "80"}
+)
+
+var _ = Describe("[external_dataplane] Connectivity", func() {
+	f := framework.NewFramework("ext-dataplane")
+
+	It("should be able to connect from an external app to a pod in a cluster", func() {
+		testExternalConnectivity(f)
+	})
+})
+
+func testExternalConnectivity(f *framework.Framework) {
+	for idx := range framework.KubeClients {
+		clusterName := framework.TestContext.ClusterIDs[idx]
+
+		By(fmt.Sprintf("Creating a pod and a service in cluster %q", clusterName))
+		np := f.NewNetworkPod(&framework.NetworkPodConfig{
+			Type:          framework.CustomPod,
+			Port:          80,
+			Cluster:       framework.ClusterIndex(idx),
+			Scheduling:    framework.NonGatewayNode,
+			ContainerName: testContainerName,
+			ImageName:     testImage,
+			Command:       simpleHTTPServerCommand,
+		})
+		svc := np.CreateService()
+
+		// Get handle for existing docker
+		docker := framework.New(extAppName)
+
+		// Get IPs to use later
+		podIP := np.Pod.Status.PodIP
+		svcIP := svc.Spec.ClusterIP
+		dockerIP := docker.GetIP(extNetName)
+
+		By(fmt.Sprintf("Sending an http request from external app %q to the service %q in the cluster %q",
+			dockerIP, svcIP, clusterName))
+		command := []string{"curl", "-m", "10", fmt.Sprintf("%s:%d", svcIP, 80)}
+		_, _ = docker.RunCommand(command...)
+
+		By("Verifying that pod got request")
+		podLog := np.GetLog()
+
+		// TODO: also verify cluster1 case
+		if clusterName != "cluster1" {
```

Comment:
We shouldn't assume a specific cluster name. These tests can be run against any cluster env, not just our CI env. I'm not really clear on the intent here but perhaps you could check `clusterName != framework.TestContext.ClusterIDs[0]` or more simply `idx != 0`?